docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md

```diff
@@ -450,3 +459,110 @@ To enable PROXY protocol support, the openstack-cloud-controller-manager config
    Request Body:
            -no body in request-
    ```
+
+### Sharing load balancer with multiple Services
+
+By default, different Services of LoadBalancer type should have different corresponding cloud load balancers, however, openstack-cloud-controller-manager allows multiple Services to share a single load balancer if the Octavia service supports the tag feature (since version 2.5).
+
+The shared load balancer can be created either by other Services or outside the cluster, e.g. created manually by the user in the cloud or by Services from the other Kubernetes clusters. The load balancer is deleted only when the last attached Service is deleted, unless the load balancer was created outside the Kubernetes cluster.
+
+The maximum number of Services that share a load balancer can be configured in `[LoadBalancer] max-shared-lb`, default value is 2. The ports of those Services shouldn't have collisions.
+
+For example, create a Service `service-1` as before:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: service-1
+  namespace: default
+spec:
+  type: LoadBalancer
+  selector:
+    app: webserver
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 8080
+```
+
+When `service-1` is created successfully, check the load balancer created in the cloud, which has its name in its tags.
+
+```shell
+$ openstack loadbalancer show 2b224530-9414-4302-8163-5abebdcdc84f -c name -c tags
++-------+---------------------------------------------+
+| Field | Value                                       |
++-------+---------------------------------------------+
+| name  | kube_service_cluster-name_default_service-1 |
+| tags  | kube_service_cluster-name_default_service-1 |
++-------+---------------------------------------------+
+```
+
+Check the Service, you should notice a new annotation `loadbalancer.openstack.org/load-balancer-id` is added:
+
+```shell
+$ kubectl describe service service-1 | grep loadbalancer.openstack.org/load-balancer-id
+                          loadbalancer.openstack.org/load-balancer-id: 2b224530-9414-4302-8163-5abebdcdc84f
+```
+
+> NOTE: Do not update the annotation `loadbalancer.openstack.org/load-balancer-id` after the Service is created successfully or the relationship between Service and the load balancer will be broken.
+
+Now, create another Service `service-2` but re-use the load balancer created for `service-1` by specifying the annotation `loadbalancer.openstack.org/load-balancer-id`:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: service-2
+  namespace: default
+  annotations:
+    loadbalancer.openstack.org/load-balancer-id: "2b224530-9414-4302-8163-5abebdcdc84f"
```

Comment:
just curious, if the ID is given but the LB not found or in some race condition like
is deleting, what will happen?

Reply:
If the load balancer is not found or in deleting status, service creation will fail.